src/main/java/io/stargate/sgv2/jsonapi/api/model/command/table/definition/datatype/ComplexTypes.java

```diff
@@ -0,0 +1,73 @@
+package io.stargate.sgv2.jsonapi.api.model.command.table.definition.datatype;
+
+import io.stargate.sgv2.jsonapi.service.schema.tables.ApiDataType;
+import io.stargate.sgv2.jsonapi.service.schema.tables.ComplexApiDataType;
+import io.stargate.sgv2.jsonapi.service.schema.tables.PrimitiveApiDataType;
+
+/** Interface for complex column types like collections */
+public interface ComplexTypes extends ColumnType {
+
+  /** A map type implimentation */
+  public static class MapType implements ComplexTypes {
+    private final ColumnType keyType;
+    private final ColumnType valueType;
+
+    public MapType(ColumnType keyType, ColumnType valueType) {
```

Comment:
Why don't we have `getKeyType()` and `getValueType()` accessors?

Reply:
I never needed that value anywhere outside so far, because we create ApiDataType using those. THought of adding it as needed.